zerver/tests/test_markdown.py

```diff
@@ -1500,12 +1500,43 @@ def test_multiple_matching_realm_patterns(self) -> None:
             converted.rendered_content,
             '<p>We should fix <a href="https://trac.example.com/ticket/ABC-123">ABC-123</a> or <a href="https://trac.example.com/ticket/16">trac ABC-123</a> today.</p>',
         )
-        # But both the links should be generated in topics.
+        # Only the older linkifier should be used in the topic, because the two patterns overlap.
         self.assertEqual(
             converted_topic,
             [
                 {"url": "https://trac.example.com/ticket/ABC-123", "text": "ABC-123"},
-                {"url": "https://other-trac.example.com/ticket/ABC-123", "text": "ABC-123"},
+            ],
+        )
+
+        url_format_string = r"https://yet-another-trac.example.com/ticket/%(id)s"
+        linkifier_3 = RealmFilter(
+            realm=realm,
+            pattern=r"(?P<id>[A-Z][A-Z0-9]+)",
```

Comment:
Can you add a comment explaining the purpose of this specific construction? I'm not entirely sure what the intent is behind it not having a `-`, for example, and it might be worth reminding a reader of how we look for word boundaries.

Reply:
Updated.